src/Query/Builder.php

```diff
@@ -1086,28 +1090,33 @@ protected function compileWhereBasic(array $where): array
             $regex = preg_replace('#(^|[^\\\])%#', '$1.*', preg_quote($value));
 
             // Convert like to regular expression.
-            if (! Str::startsWith($value, '%')) {
+            if (! str_starts_with($value, '%')) {
                 $regex = '^'.$regex;
             }
-            if (! Str::endsWith($value, '%')) {
+            if (! str_ends_with($value, '%')) {
                 $regex .= '$';
             }
 
-            $value = new Regex($regex, 'i');
-        } // Manipulate regexp operations.
-        elseif (in_array($operator, ['regexp', 'not regexp', 'regex', 'not regex'])) {
+            $value = new Regex($regex, $flags);
+        } // Manipulate regex operations.
+        elseif (in_array($operator, ['regex', 'not regex'])) {
             // Automatically convert regular expression strings to Regex objects.
             if (! $value instanceof Regex) {
                 $e = explode('/', $value);
-                $flag = end($e);
-                $regstr = substr($value, 1, -(strlen($flag) + 1));
-                $value = new Regex($regstr, $flag);
+                if ($value[0] !== '/' || count($e) < 3) {
+                    throw new \InvalidArgumentException(sprintf('Regular expressions must be surrounded by slashes "/". Got "%s"', $value));
+                }
+                $flags = end($e);
+                $regstr = substr($value, 1, -1 - strlen($flags));
+                $value = new Regex($regstr, $flags);
```

Comment:
I don't know if you want to bother validating flags (drivers typically do not, but I dunno about libraries), but I'll point this out anyway. If you're undecided we can always track this in a separate ticket.

----

Per [BSONspec.org](https://bsonspec.org/spec.html), the BSON regex type supports the following flags:

> Valid options are 'i' for case insensitive matching, 'm' for multiline matching, 'x' for verbose mode, 'l' to make \w, \W, etc. locale dependent, 's' for dotall mode ('.' matches everything), and 'u' to make \w, \W, etc. match unicode.

The [`$regex`](https://www.mongodb.com/docs/manual/reference/operator/query/regex/#mongodb-query-op.-options) operator, which isn't relevant here, supports a subset: "imxs".

Reply:
Reimplementing a validation that is already done by the server is harmful, as server rules could be changed.

Reply:
At the time I wrote that comment, I wasn't aware if the server validated the options string and has some recollection of it not doing so in old versions; however, I just tested this and it does raise an error (e.g. "invalid flag in regex options: l").

OK to defer to the server to handle this.